src/Search.elm

```diff
@@ -435,253 +537,400 @@ view :
+        ]
 
-                                Http.BadStatus code ->
-                                    ( "Bad Status", "Server returned " ++ String.fromInt code )
 
-                                Http.BadBody text ->
-                                    ( "Bad Body", text )
-                    in
-                    [ div [ class "alert alert-error" ]
-                        [ h4 [] [ text errorTitle ]
-                        , text errorMessage
-                        ]
+viewSearchInput :
+    (Msg a b -> c)
+    -> String
+    -> String
+    -> Maybe String
+    -> Html c
+viewSearchInput outMsg categoryName selectedChannel searchQuery =
+    form
+        [ onSubmit (outMsg QueryInputSubmit)
+        , class "search-input"
+        ]
+        [ div []
+            [ div []
+                [ input
+                    [ type_ "text"
+                    , id "search-query-input"
+                    , autofocus True
+                    , placeholder <| "Search for " ++ categoryName
+                    , onInput (outMsg << QueryInput)
+                    , value <| Maybe.withDefault "" searchQuery
                     ]
+                    []
+                ]
+            , button [ class "btn", type_ "submit" ]
+                [ text "Search" ]
+            ]
+        , div [] (viewChannels outMsg selectedChannel)
         ]
 
 
-viewSortSelection : (Msg a -> b) -> Model a -> Html b
-viewSortSelection outMsg model =
-    form [ class "form-horizontal pull-right sort-form" ]
-        [ div [ class "control-group sort-group" ]
-            [ label [ class "control-label" ] [ text "Sort by:" ]
+viewChannels :
+    (Msg a b -> c)
+    -> String
+    -> List (Html c)
+viewChannels outMsg selectedChannel =
+    List.append
+        [ div []
+            [ h4 [] [ text "Channel: " ]
             , div
-                [ class "controls" ]
-                [ select
-                    [ onInput (\x -> outMsg (SortChange x))
+                [ class "btn-group"
+                , attribute "data-toggle" "buttons-radio"
+                ]
+                (List.filterMap
+                    (\channelId ->
+                        channelDetailsFromId channelId
+                            |> Maybe.map
+                                (\channel ->
+                                    button
+                                        [ type_ "button"
+                                        , classList
+                                            [ ( "btn", True )
+                                            , ( "active", channel.id == selectedChannel )
+                                            ]
+                                        , onClick <| outMsg (ChannelChange channel.id)
+                                        ]
+                                        [ text channel.title ]
+                                )
+                    )
+                    channels
+                )
+            ]
+        ]
```

Comment:
I can't put suggestion in but this can be simplified to using `cons` on a list instead.

`(div [] .....) :: (List.fiterMap)`